src/wp-includes/plugin.php

```diff
@@ -773,16 +773,16 @@ function plugin_basename( $file ) {
 	foreach ( $wp_plugin_paths as $dir => $realdir ) {
 		if ( str_starts_with( $file, $realdir ) ) {
 			$file = $dir . substr( $file, strlen( $realdir ) );
+			break;
 		}
 	}
 
 	$plugin_dir    = wp_normalize_path( WP_PLUGIN_DIR );
 	$mu_plugin_dir = wp_normalize_path( WPMU_PLUGIN_DIR );
 
 	// Get relative path from plugins directory.
-	$file = preg_replace( '#^' . preg_quote( $plugin_dir, '#' ) . '/|^' . preg_quote( $mu_plugin_dir, '#' ) . '/#', '', $file );
-	$file = trim( $file, '/' );
-	return $file;
+	$file = preg_replace( '#^(?>' . preg_quote( $plugin_dir, '#' ) . '|' . preg_quote( $mu_plugin_dir, '#' ) . ')/#', '', $file, 1 );
+	return trim( $file, '/' );
```

Comment:
```suggestion
	$file = trim( $file, '/' );
	return $file;
```

Let's not change this if it's not necessary.

Reply:
Why keep a bad practice?